### ISSUE TITLE
Use UnsafeAccessorAttribute in dotnet8 instead of reflection

### ISSUE THREAD

**Issue by Genmutant** [m31coding/M31.FluentAPI#16]
Would it be possible to generate private method accessors using [UnsafeAccessorAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafeaccessorattribute?view=net-8.0) when targetting at least dotnet8?
That would make it safe to use with NativeAOT and Trimming, as the accessed attributes are marked as used and not removed.

**Comment by m31coding** [comment]
Thank you for pointing this out! The `UnsafeAccessorAttribute` is indeed a great feature, and I would love to eliminate the reflection code. However, I plan to stick with .NET 6 until the end of the year, and I prefer not to complicate the code with conditional compilation directives (ifdefs).

Is there another way to address your issue? Are there possibilities for configuring the trimming, such as ignoring a specific class during the trimming process?

Best regards,
Kevin

**Comment by Genmutant** [comment]
Yes, it's possible to manually force it to keep the complete classes when trimming. I haven't checked if AOT would still work though.

Thanks for your fast answer and work!

**Comment by m31coding** [comment]
Hi @Genmutant,

I have finally implemented your proposal! I use .NET 10 instead of .NET 8 because only the former supports UnsafeAccessors with generics.

Thank you for your contribution!

Best regards,
Kevin
